tests/model_registry/rbac/conftest.py

```diff
@@ -89,6 +92,77 @@ def sa_token(service_account: ServiceAccount) -> str:
         raise
 
 
+@pytest.fixture(scope="function")
+def new_group(
+    request: pytest.FixtureRequest,
+    admin_client: DynamicClient,
+    test_idp_user_session: UserTestSession,
+) -> Generator[str, None, None]:
+    """
+    Fixture to create a new OpenShift group and add a user, then delete the group after the test.
+    The group name is passed as a parameter to the fixture and the user is taken from test_idp_user_session.
+    """
+
+    group_name = request.param
+    with Group(
+        client=admin_client,
+        name=group_name,
+        users=[test_idp_user_session.username],
+        wait_for_resource=True,
+    ) as _:
+        LOGGER.info(f"Group {group_name} created successfully.")
+        yield group_name
+        LOGGER.info(f"Group {group_name} deletion initiated by context manager.")
+
+
+@pytest.fixture(scope="function")
+def model_registry_group_with_user(
+    admin_client: DynamicClient,
+    test_idp_user_session: UserTestSession,
+) -> Generator[Group, None, None]:
+    """
+    Fixture to manage a test user in the Model Registry group.
+    Adds the user to the group before the test, then removes them after.
+
+    Args:
+        admin_client: The admin client for accessing the cluster
+        test_idp_user_session: The test user session containing user information
+
+    Yields:
+        Group: The Model Registry group with the test user added
+    """
+    model_registry_users_group = f"{MR_INSTANCE_NAME}-users"
+    group = Group(
+        client=admin_client,
+        name=model_registry_users_group,
+        wait_for_resource=True,
+    )
+
+    # Add user to group
+    group.update(
+        resource_dict={"metadata": {"name": model_registry_users_group}, "users": [test_idp_user_session.username]}
+    )
+    LOGGER.info(f"Added user {test_idp_user_session.username} to {model_registry_users_group} group")
```

Comment:
You can use ResourceEditor here. That way you won't need to keep track of the clean up.